cli/src/commands/schema/edit.ts

```diff
@@ -1,107 +1,64 @@
-import { getBranchDetailsWithPgRoll } from '../../migrations/pgroll.js';
-import { isNil, reportBugURL } from '../../utils.js';
-import Codegen from '../codegen/index.js';
-import Pull from '../pull/index.js';
+import {
+  AddColumnPayload,
+  AddTablePayload,
+  ColumnAdditions,
+  ColumnData,
+  ColumnEdits,
+  DeleteColumnPayload,
+  DeleteTablePayload,
+  EditColumnPayload,
+  EditTablePayload,
+  SelectChoice,
+  ValidationState
+} from './types.js';
+import {
+  exhaustiveCheck,
+  generateLinkReference,
+  getBranchDetailsWithPgRoll,
+  requiresUpArgument,
+  waitForMigrationToFinish,
+  xataColumnTypeToPgRoll,
+  xataColumnTypeToPgRollComment,
+  xataColumnTypeToPgRollConstraint,
+  xataColumnTypeToZeroValue
+} from '../../migrations/pgroll.js';
 
-// The enquirer library has type definitions but they are very poor
 const { Select, Snippet, Confirm } = enquirer as any;
 
-type Schema = Schemas.Schema;
-type Table = Schema['tables'][0];
-type Column = Table['columns'][0];
-
-type EditableColumn = Column & {
-  added?: boolean;
-  deleted?: boolean;
-  initialName?: string;
-  description?: string;
-};
-
-type EditableTable = Table & {
-  added?: string;
-  deleted?: boolean;
-  initialName?: string;
-  columns: EditableColumn[];
-};
-
-type ColumnEditState = {
-  initial: {
-    name: string;
-    type: string;
-    link: string | undefined;
-    vectorDimension: string | undefined;
-    notNull: string;
-    defaultValue: string;
-    unique: string;
-    description: string | undefined;
-  };
-  values: {
-    name?: string;
-    type?: string;
-    link?: string;
-    vectorDimension?: string;
-    notNull?: string;
-    defaultValue?: string;
-    unique?: string;
-    description?: string;
-  };
-};
-
-const types = ['string', 'int', 'float', 'bool', 'text', 'multiple', 'link', 'email', 'datetime', 'vector', 'json'];
-const typesList = types.join(', ');
-const identifier = /^[a-zA-Z0-9-_~]+$/;
-
-const uniqueUnsupportedTypes = ['text', 'multiple', 'vector', 'json'];
-const defaultValueUnsupportedTypes = ['multiple', 'link', 'vector'];
-const notNullUnsupportedTypes = defaultValueUnsupportedTypes;
-
-const waitFlags: Record<string, string> = {
-  code: '-w',
-  'code-insiders': '-w',
-  vscodium: '-w',
-  sublime: '-w',
-  textmate: '-w',
-  atom: '--wait',
-  webstorm: '--wait',
-  intellij: '--wait',
-  xcode: '-w'
-};
-
-type SelectChoice = {
-  name:
-    | {
-        type: 'space' | 'schema' | 'add-table' | 'migrate';
-      }
-    | {
-        type: 'add-column' | 'edit-table';
-        table: EditableTable;
-      }
-    | {
-        type: 'edit-column';
-        table: EditableTable;
-        column: EditableColumn;
-      };
-  message: string;
-  role?: string;
-  choices?: SelectChoice[];
-  disabled?: boolean;
-  hint?: string;
-};
+const xataTypes = [
+  'string',
+  'int',
+  'float',
+  'bool',
+  'text',
+  'multiple',
+  'link',
+  'email',
+  'datetime',
+  'vector',
+  'json',
+  'file',
+  'file[]'
+];
```

Comment:
Even if we don't expose more postgres types for now, let's stop using xata types and move to Postgres ones.